drgn_tools/meminfo.py

```diff
@@ -0,0 +1,678 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helpers for dumping memory usage information and statistics
+"""
+import argparse
+import math
+from collections import UserDict
+from typing import Dict
+from typing import Iterator
+from typing import List
+
+import drgn
+from drgn import Object
+from drgn import Program
+from drgn.helpers.linux import list_for_each_entry
+from drgn.helpers.linux.percpu import percpu_counter_sum
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.mm import totalram_pages
+from drgn_tools.util import has_member
+
+
+__all__ = ("show_all_meminfo", "get_all_meminfo")
+
+
+class StatDict(UserDict):
+    """A dictionary wrapper class to handle renaming of statistics items."""
+
+    renames = {
+        "NR_SLAB_RECLAIMABLE": "NR_SLAB_RECLAIMABLE_B",
+        "NR_SLAB_UNRECLAIMABLE": "NR_SLAB_UNRECLAIMABLE_B",
+        "NR_ANON_MAPPED": "NR_ANON_PAGES",
+        "NR_KERNEL_STACK_KB": "NR_KERNEL_STACK",
+    }
+
+    def __getitem__(self, key):
+        """
+        Some global statistics items's names was changed in the past. This
+        overwrites the default __getitem__ function to correct name for these.
+        """
+        if key in self.renames and key not in self.data:
+            key = self.renames[key]
+        return super().__getitem__(key)
+
+
+def _read_stats(prog: Program, arr_name: str, enum: str) -> UserDict:
+    """
+    Read counters from a global statistics array with name ```arr_name```.
+    Parse all statistics items that are indexed by ```enum``` and store results
```

Comment:
nit: use two backticks for the inline code:
```suggestion
    Read counters from a global statistics array with name ``arr_name``.
    Parse all statistics items that are indexed by ``enum`` and store results
```
this applies everywhere and should be a quick fix. It's restructured text format, in case you're wondering.

Reply:
Sounds great. Thanks for pointing it out.